src/dmd/ob.d

```diff
@@ -1976,7 +1976,12 @@ void checkObErrors(ref ObState obstate)
             else if (isReadonlyPtr(v))
                 pvs.state = PtrState.Readonly;
             else
+            {
+                if (pvs.state == PtrState.Owner && v.type.hasPointersToMutableFields())
+                    v.error(e.loc, "assigning to Owner without disposing of owned value");
```

Comment:
This error message could be improved, how about 
```
is assigned to Owner %s (which?) without disposing of owned value %s (what?)
```

Reply:
ping

Reply:
Also, a supplemental error, hinting at how we can workaround this error might be helpful.